filesec/file_security_test.go

```diff
@@ -505,6 +507,48 @@ var _ = Describe("FileSSL", func() {
 			Expect(stat.Size()).To(Equal(int64(16)))
 		})
 
+		It("Should always overwrite files", func() {
+			c, err := config.NewDefaultConfig()
+			Expect(err).ToNot(HaveOccurred())
+
+			identity := "rip.mcollective"
+
+			// These certs both have the same hostname.  First we cache the first one, then we attempt to cache the second one.
+			// This should result in the caching layer storing the second certificate.
+			firstcert := filepath.Join("..", "testdata", "intermediate", "certs", identity+".pem")
+			secondcert := filepath.Join("..", "testdata", "intermediate", "certs", "second."+identity+".pem")
+
+			c.Choria.FileSecurityCertificate = firstcert
+			c.Choria.FileSecurityCA = filepath.Join("..", "testdata", "intermediate", "certs", "ca.pem")
+			c.Choria.FileSecurityCache = filepath.Join("..", "testdata", "intermediate", "certs")
+			c.Choria.SecurityAlwaysOverwriteCache = true
+
+			c.Choria.FileSecurityCache, err = ioutil.TempDir("", "cache-always")
+			Expect(err).ToNot(HaveOccurred())
+			defer os.RemoveAll(c.Choria.FileSecurityCache)
+
+			prov, err := New(WithChoriaConfig(c), WithLog(l.WithFields(logrus.Fields{})))
+			Expect(err).ToNot(HaveOccurred())
+
+			fpd, err := ioutil.ReadFile(firstcert)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = prov.CachePublicData(fpd, identity)
+			Expect(err).ToNot(HaveOccurred())
+
+			spd, err := ioutil.ReadFile(secondcert)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = prov.CachePublicData(spd, identity)
+			Expect(err).To(BeNil())
+
+			cpd, err := prov.CachedPublicData(identity)
+			Expect(err).ToNot(HaveOccurred())
+
+			res := bytes.Compare(spd, cpd)
+			Expect(res).To(BeZero())
+		})
```

Comment:
hmm, did we need all these new files and test data, i mean you could simply take a cert, write it to a temp directory cache and record its mtime, then attempt to cache again - if mtime change the cache updated? anyway no biggie this works too - mtime probably is 1 second resolution which would be annoying in tests

Reply:
@ripienaar I didn't keep the CA private keys around >.<  so had to regenerate them.

I figure the byte slices would be small enough that it wouldn't matter in a test.